snapcraft/plugins/cmake.py

```diff
@@ -64,22 +64,15 @@ def __init__(self, name, options, project):
         self.build_packages.append('cmake')
 
     def build(self):
-        if os.path.exists(self.builddir):
-            shutil.rmtree(self.builddir)
-        os.mkdir(self.builddir)
```

Comment:
Because the pluginhandler copies the source to the builddir before handing control off to the plugin. Since cmake knows it's about to do an out-of-source build using the sourcedir, it empties the builddir so it has a clean area in which to build.